### PR TITLE
deps: Update Go SDK to 54.0.0

### PR DESCRIPTION
**Pull request by favoretti**

### CODE REVIEW

File: azurerm/internal/services/resource/resource_group_resource.go [hunk 1]
```diff
@@ -124,7 +124,8 @@ func resourceResourceGroupDelete(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
-	deleteFuture, err := client.Delete(ctx, id.ResourceGroup)
+	// @favoretti: This is currently the only supported added value, do we want to add it as an optional parameter to the schema?
+	deleteFuture, err := client.Delete(ctx, id.ResourceGroup, "Microsoft.Compute/virtualMachines,Microsoft.Compute/virtualMachineScaleSets")
```
Comment (favoretti) [inline azurerm/internal/services/resource/resource_group_resource.go:128]: @katbyte @tombuildsstuff Opinions on how to go about this one?

Comment (jackofallops) [reply azurerm/internal/services/resource/resource_group_resource.go:128]: Just taken a quick look and for maintaining existing behaviour I think this should to be:
```suggestion
	deleteFuture, err := client.Delete(ctx, id.ResourceGroup, "")
```

Comment (favoretti) [reply azurerm/internal/services/resource/resource_group_resource.go]: @jackofallops thank you, commited suggestion. Wasn't sure whether we want it as an empty string or not.
I think we can extrapolate that into a separate argument once this gets in?

**Comment by favoretti** [comment]
Rebase, rename file introduced in #11552

**Comment by favoretti** [comment]
@jackofallops this should now be good to go, if you could take another look please?

**Review by katbyte — APPROVED** [review]
LGTM 👍

**Pull request merged**